src/Number.php

```diff
@@ -117,7 +117,7 @@ public static function fromString($number)
     /**
      * @param float $floatingPoint
      *
-     * @return Number
+     * @return self
```

Comment:
I tend to prefer using the class name in return type hints to indicate a new object is going to be returned. At least for consistency, this should be reverted.

Reply:
i was making consistent with `fromString` above. i'll revert this and you can handle in another PR if you wish.